test/pgw_SUITE.erl

```diff
@@ -3817,3 +3845,23 @@ set_online_charging(Set) ->
     {ok, Cfg0} = application:get_env(ergw, charging),
     Cfg = set_online_charging(['_', rulebase, '_'], Set, Cfg0),
     ok = application:set_env(ergw, charging, Cfg).
+
+%% Set to default of 8 hours (28800000 ms) after Idle-Timeout
+set_def_timeout(K, Value, APNs) ->
+    Value2  = maps:put('Idle-Timeout', 28800000, Value), % 8 hours
+    APNs#{K => Value2}.
+
+%% Set shorter timeout of 2000 ms for test purposes.
+set_new_timeout(K, Value, APNs) ->
+    Value2  = maps:put('Idle-Timeout', 2000, Value),
+    APNs#{K => Value2}.
+
+set_idle_timeout(Tmr_lth) ->
+    {ok, APNs0} = application:get_env(ergw, apns),
+    case Tmr_lth of
+	default ->
+	    APNs = maps:fold(fun set_def_timeout/3, #{}, APNs0);
+	_ ->
+	    APNs = maps:fold(fun set_new_timeout/3, #{}, APNs0)
+    end,
+    ok = application:set_env(ergw, apns, APNs).
```

Comment:
can be made simple with maps:update_with/4. Also by using a additional argument for set_*_timeout it will become even simpler, e.g.:

```
set_apn_key(Key, Value) ->
    {ok, APNs0} = application:get_env(ergw, apns),
    Upd = fun(APN) -> maps:update_with(Key, fun(_) -> Value end, Value, APN) end,
    APNs = maps:map(Upd, APNs0),
    ok = application:set_env(ergw, apns, APNs).
```

And then use a ```set_apn_key('Idle-Timeout', 300)``` in your test case.

Reply:
Ok cool! Thanks for the tip, Appreciated. will update accordingly.